### ISSUE TITLE
After exiting app with back button Platform views on Android 14 turn blank on resume.

### ISSUE THREAD

**Issue by mr-pant** [flutter/flutter#148662]
### Steps to reproduce

The issue looks very similar to https://github.com/flutter/flutter/issues/146499 but the steps are different. 
This is happening with Flutter 3.22.0

**Environment**: Samsung Galaxy A53 5G (Android 14)

**Steps:**
1. Install the app and launch.
2. From root page (with platform view), press the back button key to move app to background. 
3. Launch the app again either from launcher or from icon.

### Expected results

Platform view should be rendered at home page.

### Actual results

Platform view turns blank at home page.

Further, after the issue is reproduced, if you switch to another app from recent apps launcher , and then switch back again to the sample app, the platform view starts to appear again.

### Code sample

Sample is same as #146499:
https://github.com/flutter/flutter/files/14917706/platform_view_test.3.zip

<details open><summary>Code sample</summary>

Dart:
```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
      final Map<String, dynamic> creationParams = <String, dynamic>{};

    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text("App"),
      ),
      body: AndroidView(
        viewType: "nativeView",
        layoutDirection: TextDirection.ltr,
        creationParams: creationParams,
        creationParamsCodec: const StandardMessageCodec(),
      ),
    );
  }
}
```

Platform side:
```dart
class MyApplication : Application() {
    private lateinit var flutterEngine : FlutterEngine

    override fun onCreate() {
        super.onCreate()

        // Instantiate a FlutterEngine.
        flutterEngine = FlutterEngine(this)

        // Start executing Dart code to pre-warm the FlutterEngine.
        flutterEngine.dartExecutor.executeDartEntrypoint(
            DartExecutor.DartEntrypoint.createDefault()
        )

        // Cache the FlutterEngine to be used by FlutterActivity.
        FlutterEngineCache
            .getInstance()
            .put("my_engine_id", flutterEngine)
    }
}

class RootActivity: Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        startActivity(
            FlutterActivity.CachedEngineIntentBuilder(MainActivity::class.java, "my_engine_id")
                .build(this)
        )
        finish()
    }
}

class MainActivity: FlutterActivity() {
    override fun configureFlutterEngine(flutterEngine: FlutterEngine) {
        super.configureFlutterEngine(flutterEngine)
        flutterEngine
            .platformViewsController
            .registry
            .registerViewFactory("nativeView", NativeViewFactory())
    }
}
```

</details>

### Screenshots or Video

<details open>
<summary> Video demonstration</summary>

https://github.com/flutter/flutter/assets/14113872/eb06c121-c89b-4eea-b1bb-062d244c63d6

### Flutter Doctor output

```console
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.4.1 23E224 darwin-arm64, locale en-IN)
    • Flutter version 3.22.0 on channel stable at /Users/rpant/Flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (11 days ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/rpant/Library/Android/sdk
    • Platform android-34, build-tools 33.0.2
    • ANDROID_HOME = /Users/rpant/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C65
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.88.0

[✓] Connected device (5 available)
    • SM A536E (mobile)               • RZCTC0B6D9L                          • android-arm64  •
      Android 14 (API 34)
    • iPhone 15 Pro (mobile)          • 99935264-E514-44AE-AC05-D0ADDC217EDB • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-17-2 (simulator)
    • macOS (desktop)                 • macos                                • darwin-arm64   •
      macOS 14.4.1 23E224 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         •
      macOS 14.4.1 23E224 darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript •
      Google Chrome 125.0.6422.60

[✓] Network resources
    • All expected network resources are available.

• No issues found!```

**Comment by darshankawar** [comment]
> The issue looks very similar to #146499 but the steps are different.

@mr-pant This issue was reported on Pixel devices which was fixed. Since you are seeing the issue on Samsung, a similar [issue](https://github.com/flutter/flutter/issues/139630) related to it was fixed at Samsung side itself that you can check and confirm at your end.

**Comment by mr-pant** [comment]
> a similar [issue](https://github.com/flutter/flutter/issues/139630) related to it was fixed at Samsung side itself that you can check and confirm at your end.

@darshankawar , the mentioned issue is different in which the entire app stops drawing on resume. This issue is just with the platform view as you can see that the app bar is visible. Also it recovers after switch from other app. Which does not happen in the other issue. Let me know if you need any other detail to differentiate the issue.

**Comment by darshankawar** [comment]
Thanks for the update @mr-pant. If you see the linked issue, it was solved at Samsung's end. Is there a way for you to check the same behavior on device other than Samsung (ex: Pixel) just to narrow down the issue is Samsung specific or not ?


> This is happening with Flutter 3.22.0

If you roll back to 3.19.6 and re-run your scenario, you don't see the reported error ?

**Comment by johnmccutchan** [comment]
I just tried reproducing on Android 14 using Flutter master.

See attached video:

https://github.com/flutter/flutter/assets/224266/d7170304-f95b-4b6e-96b6-5026c232cdc6

**Comment by johnmccutchan** [comment]
I also could not reproduce on 3.22:

```
$ flutter --version
Flutter 3.22.0 • channel stable • git@github.com:flutter/flutter.git
Framework • revision 5dcb86f68f (12 days ago) • 2024-05-09 07:39:20 -0500
Engine • revision f6344b75dc
Tools • Dart 3.4.0 • DevTools 2.34.3
```

**Comment by johnmccutchan** [comment]
The above two tests were against an emulator. Trying on a Pixel 7 Android 14 phone.

**Comment by johnmccutchan** [comment]
I can reproduce on my physical device. Digging in.

**Comment by johnmccutchan** [comment]
I've reproduced the issue and have a fix (but I don't like the fix...) 

This appears to be another instance of the underlying Android bug discussed [here](https://github.com/flutter/flutter/issues/146499). 

Waiting to hear back from the Android folks on this.

**Comment by johnmccutchan** [comment]
Just documenting the difference between this bug and #146499:

#146499:
* App is launched
* App is backgrounded (via a swipe up)
* Android Bug triggers HERE
* onTrimMemory callback fires in Flutter and we work around the bug above.
* App is resumed
* Platform views render correctly

#148662:
* App is launched
* App is backgrounded (via a back-gesture swipe)
* Android Bug triggers HERE
* no onTrimMemory callback fires
* App is resumed
* Platform views don't render

So, with the back-gesture the onTrimMemory callback doesn't fire but with the up-gesture it does.

**Comment by johnmccutchan** [comment]
Note, we will include this fix in a 3.22.x point release.
